1. Basic Web Development/script.ts

```diff
@@ -0,0 +1,405 @@
+    this.BOXES = getBoxes();
+
+    handleRadios();
+}
+
+function routingRoutine() {
+    navigate(location.hash);
+
+    window.addEventListener('hashchange', () => {
+        navigate(location.hash);
+    });
+}
+
+function navigate(toLocation: string) {
+    if (toLocation == '#edit') {
+        document.getElementById("creation-page").classList.add("no-display");
+        document.getElementById("edit-page").classList.remove("no-display");
+
+        document.getElementsByClassName("section")[1].classList.add("editing");
+
+        document.getElementsByClassName("editor")[0].classList.remove('no-display');
+        document.getElementsByClassName("preferences")[0].classList.remove('presentation');
+        document.getElementsByClassName("begin-arrow")[0].classList.add('no-display');
+    } else if (toLocation == '#create') {
+        document.getElementById("creation-page").classList.remove("no-display");
+        document.getElementById("edit-page").classList.add("no-display");
+
+        document.getElementsByClassName("section")[1].classList.remove("editing");
+
+        document.getElementsByClassName("editor")[0].classList.remove('no-display');
+        document.getElementsByClassName("preferences")[0].classList.remove('presentation');
+        document.getElementsByClassName("begin-arrow")[0].classList.add('no-display');
+    } else {
+        document.getElementsByClassName("editor")[0].classList.add('no-display');
+        document.getElementsByClassName("preferences")[0].classList.add('presentation');
+        document.getElementsByClassName("begin-arrow")[0].classList.remove('no-display');
+    }
+}
+
+function getBoxes() {
+    let pmsBoxes = document.querySelectorAll('input[name="permissionPms"]') as NodeListOf<HTMLInputElement>;
+    let allBoxes = document.querySelectorAll('input[name="permissionAll"]') as NodeListOf<HTMLInputElement>;
+
+    return { pmsBoxes, allBoxes };
+}
+
+function handleModification(form: HTMLElement) {
+    document.addEventListener('click', event => {
+        const target = event.target as HTMLInputElement
+        if (target.classList.contains('delete-btn') && target.classList.contains('edit-btn')) {
+            document.getElementById('edit-form').classList.remove('no-display');
+
+            const elemenToEdit: Permission = this.PERMISSIONS.filter((perm: Permission) => perm.id == target.name)[0];
+            this.NOWEDITING = elemenToEdit;
+
+            document.getElementById(elemenToEdit.type + 'Edit').click();
+
+            (document.getElementById('permNameEdit') as HTMLInputElement).value = elemenToEdit.text;
+            createPermissionsForm(form);
+        } else if (target.classList.contains('delete-btn')) {
+            this.PERMISSIONS = this.PERMISSIONS.filter((el: Permission) => el.id != target.name);
+            createPermissionsForm(form);
+
+            fetch(API_URL + '/pref/delete', { method: 'DELETE', headers: HEADERS, body: JSON.stringify({name: target.name}) })
```

Comment:
You don't handle the case in which you have selected an item to edit, and then you delete it. You still see the edit mode of the item that you've already deleted.